### ISSUE TITLE
does the types of neuron include both excitatory and inhibitory?

### ISSUE THREAD

**Issue by 201528014227051** [fangwei123456/spikingjelly#240]
does the types of neuron include both excitatory and inhibitory?

**Comment by fangwei123456** [comment]
No. If you want to implement the inhibitory neuron, you can use a ±1 mask to multiply the output spikes of the neurons layer.

**Comment by tmasquelier** [comment]
I think separating excitatory and inhibitory neurons makes no sense if you don't respect Dale's law: inhibitory (respectively excitatory) neurons only have negative (respectively positive) output weights.
Here is an example:
https://arxiv.org/abs/2206.10226
and the corresponding PyTorch code:
https://github.com/fmi-basel/stork

**Comment by 201528014227051** [comment]
Thanks



王斌强
中国科学院，西安光学精密机械研究所




&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fangwei123456/spikingjelly"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月30日(星期四) 下午3:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fangwei123456/spikingjelly] does the types of neuron include both excitatory and inhibitory? (Issue #240)





 
No. If you want to implement the inhibitory neuron, you can use a ±1 mask to multiply the output spikes of the neurons layer.

**Comment by 201528014227051** [comment]
Thanks a lot
Can I hold the opinion that the trianed weights maybe include both the excitatory and inhibitory? Because the weight can be both positive or negative during training.



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fangwei123456/spikingjelly"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月30日(星期四) 下午3:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fangwei123456/spikingjelly] does the types of neuron include both excitatory and inhibitory? (Issue #240)





 
I think separating excitatory and inhibitory neurons makes no sense if you don't respect Dale's law: inhibitory (respectively excitatory) neurons only have negative (respectively positive) output weights.
 Here is an example:
 https://arxiv.org/abs/2206.10226
 and the corresponding PyTorch code:
 https://github.com/fmi-basel/stork

**Comment by fangwei123456** [comment]
> Can I hold the opinion that the trianed weights maybe include both the excitatory and inhibitory?
Yes. If you want to specify the  excitatory and inhibitory weights, you can use `weight.abs() * mask`.

**Comment by 201528014227051** [comment]
Thanks



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fangwei123456/spikingjelly"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月1日(星期五) 上午9:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fangwei123456/spikingjelly] does the types of neuron include both excitatory and inhibitory? (Issue #240)





  
Can I hold the opinion that the trianed weights maybe include both the excitatory and inhibitory?
 Yes. If you want to specify the  excitatory and inhibitory weights, you can use weight.abs() * mask.
